.github/workflows/e2e_libvirt.yaml

```diff
@@ -35,8 +45,102 @@ jobs:
+          # file directly.
+          sudo setfacl -m "u:${USER}:rwx" /var/run/libvirt/libvirt-sock
+
+      - name: Install kcli
+        run: |
+          echo "::group::Install dependencies"
+          sudo apt-get install -y genisoimage
+          echo "::endgroup::"
+
+          echo "::group::Install kcli"
+          curl https://raw.githubusercontent.com/karmab/kcli/main/install.sh | sudo bash
+          echo "::endgroup::"
+
+          # kcli needs a pair of keys to setup the VMs
+          [ -f ~/.ssh/id_rsa ] || \
+            ssh-keygen -t rsa -f ~/.ssh/id_rsa -N ""
+
+          # Newest version of kcli does not index old Ubuntu images like 20.04
+          echo "::group::Download Ubuntu 20.04 image"
+          kcli download image -u https://cloud-images.ubuntu.com/releases/20.04/release/ubuntu-20.04-server-cloudimg-amd64.img ubuntu2004
+          echo "::endgroup::"
+
+        # This is the key used by cloud-api-adaptor to connect to libvirt
+      - name: Generate the SSH key
+        run: |
+          ssh-keygen -f ./id_rsa -N ""
+          mkdir -p ~/.ssh
+          cat id_rsa.pub >> ~/.ssh/authorized_keys
+          chmod 600 ~/.ssh/authorized_keys
+        working-directory: install/overlays/libvirt
+
+      - name: Verify the connection with Libvirt
+        id: verify_libvirt_connection
+        run: |
+          IP="$(hostname -I | cut -d' ' -f1)"
+          echo "ip=${IP}" >> "$GITHUB_OUTPUT"
+
+          virsh -c "qemu+ssh://$USER@${IP}/system?keyfile=$(pwd)/id_rsa&no_verify=1" nodeinfo
+        working-directory: install/overlays/libvirt
+
+      - name: Create the e2e properties file
+        run: |
+          IP=${{ steps.verify_libvirt_connection.outputs.ip }}
+          [[ -n "$IP" ]] || exit 1
+          echo "libvirt_uri=\"qemu+ssh://$USER@${IP}/system?no_verify=1\"" >> libvirt.properties
+          echo "libvirt_ssh_key_file=\"id_rsa\"" >> libvirt.properties
+          # For debugging
+          cat libvirt.properties
+
+      - name: Install tests dependencies
+        run: |
+          K8S_VERSION="v1.27.1"
+          KUSTOMIZE_VERSION="3.8.7"
+
+          if ! command -v kubectl >/dev/null; then
+            sudo curl "https://storage.googleapis.com/kubernetes-release/release/${K8S_VERSION}/bin/linux/amd64/kubectl" \
+              -o /usr/local/bin/kubectl
+            sudo chmod +x /usr/local/bin/kubectl
+          fi
+
+          if ! command -v kustomize >/dev/null; then
+          curl -s https://raw.githubusercontent.com/kubernetes-sigs/kustomize/master/hack/install_kustomize.sh | \
+            sudo bash -s -- "${KUSTOMIZE_VERSION}" /usr/local/bin
+            sudo chmod +x /usr/local/bin/kustomize
+          fi
+
+          sudo apt-get install -y build-essential
+
       - name: run tests
         run: |
-          ls podvm
-          tree install
-          git diff
+          export TEST_PROVISION="yes"
+          export TEST_TEARDOWN="no"
```

Comment:
I might come to it in a later commit, but is Teardown no so we can get debug info and the test runner node should be cleaned up anyway at the end of the workflow?

Reply:
Exactly Steve!